datafusion_iceberg/src/table.rs

```diff
@@ -561,17 +572,25 @@ async fn table_scan(
                                 last_updated_ms,
                             )?;
 
-                            let delete_file_scan_config = FileScanConfig {
-                                object_store_url: object_store_url.clone(),
-                                file_schema: delete_file_schema,
-                                file_groups: vec![vec![delete_file]],
-                                constraints: Constraints::empty(),
-                                statistics: statistics.clone(),
-                                projection: None,
-                                limit,
-                                table_partition_cols: table_partition_cols.clone(),
-                                output_ordering: vec![],
-                            };
+                            let delete_file_source =
+                                if let Some(physical_predicate) = physical_predicate.clone() {
+                                    Arc::new(ParquetSource::default().with_predicate(
+                                        Arc::clone(&delete_file_schema),
+                                        physical_predicate,
+                                    ))
+                                } else {
+                                    Arc::new(ParquetSource::default())
+                                };
```

Comment:
Check here too please

Reply:
Similar to the above, I would activate the filter pushdown.

Reply:
Added to it